Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java

```diff
@@ -154,10 +173,63 @@
 
     }
 
+
+    /**
+     * Rest interface to get conceptcollections related to workspace
+     * http://<<URL>:<PORT>>/quadriga/auth/rest/workspace/{workspaceid}/conceptcollections.json
+     * http://localhost:8080/quadriga/auth/rest/workspace/e23a8585-20bc-458e-ab7d-c758962b11aa/conceptcollections.json
+     * 
+     * 
+     * @author Ajay Modi & Bharath Srikantan
+     * @param workspaceid
+     * @param model
+     * @param principal
+     * @return
+     */
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.WORKSPACE, paramIndex = 1, userRole = { RoleNames.ROLE_WORKSPACE_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "auth/rest/workspace/{workspaceid}/conceptcollections.json", method = RequestMethod.GET, produces = "application/json")
+    @ResponseBody
+    public ResponseEntity<String> listWorkspaceConceptCollection(
+            @PathVariable("workspaceid") String workspaceId, Model model,
+            Principal principal) {
+        String userId = principal.getName();
+        
+        List<IWorkspaceConceptCollection> conceptCollectionList = null;
+        JSONArray ja = new JSONArray();
+        try {
+            conceptCollectionList = workspaceCCManager.listWorkspaceCC(
+                    workspaceId, userId);    
+    
+            if(conceptCollectionList!=null){
+                
+                for (IWorkspaceConceptCollection conceptCollection : conceptCollectionList) {
+                    JSONObject j = new JSONObject();
+                    try {
+                        j.put("id", conceptCollection.getConceptCollection()
+                                .getConceptCollectionId());
+                        j.put("name", conceptCollection.getConceptCollection()
+                                .getConceptCollectionName());
+                        ja.put(j);
+        
+                    } catch (JSONException e) {
+                        logger.error("JSONException:", e);
+                        return new ResponseEntity<String>(e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+                    }
+                }
+            }
+            
+        }catch (QuadrigaStorageException e) {

```

Comment:
this catch block should be where the exception is actually thrown (line 200 I believe).